### DISCUSSION TITLE
Login Functionality for Specific Endpoint

### DISCUSSION THREAD

**Discussion by rmuhawieh** [imartinez/privateGPT#1631] [General]
Say I made two endpoints, each with slightly different gradio blocks, and I want one to be accessible to anyone and the other to require a login/authentication. How would you create that endpoint specific login function?

**Comment by tedraymond** [comment]
In the gradio app.launch()

demo.launch(auth=("admin", "pass1234"))

**Reply by rmuhawieh** [reply]
I'm not using .launch for my gradio apps though, I am mounting them onto a fastapi app, meaning I can't use the above authentication. Do you know of a way to do it for mounting the gradio apps?
